common/src/main/java/com/graphicsfuzz/common/util/ShaderJobFileOperations.java

```diff
@@ -1186,21 +1201,80 @@ protected static void writeShaderResultToFileHelper(
             "outputs", new Gson().fromJson(shaderResult.getComputeOutputs(), JsonObject.class));
       }
 
+      // We write out the .info.json file now, so that the Python tooling for diffing compute
+      // shader results can be invoked on it if needed.
       fileOps.writeStringToFile(
-          new File(shaderJobResultNoExtension + ".info.json"),
+          shaderJobResultFile,
           infoJson.toString());
 
-      return;
-    }
+      if (referenceShaderResultFile.isPresent()) {
 
-    final File outputImage = new File(shaderJobResultNoExtension + ".png");
+        // We have reference results, so can populate the "comparison_with_reference" property.
+
+        // This maps to a dictionary with up to 4 keys:
+        // - "exact_match", true if and only if the results are identical
+        // - "exactdiff_output", populated only if "exact_match" is false, with the result of
+        //   exact diffing
+        // - "fuzzy_match", present only if "exact_match" is false, and then true if and only if
+        //   the results are similar
+        // - "fuzzydiff_output", present only if "fuzzy_diff" is set, with the result of
+        //   fuzzy diffing.
+
+        final JsonObject computeShaderComparisonWithReference = new JsonObject();
+        try {
+
+          // Check whether the results exactly match those of the reference.
+          final ExecResult exactDiffResult =
+              new ExecHelper().exec(ExecHelper.RedirectType.TO_BUFFER,
+              null, false,
+                  Paths.get(ToolPaths.getPythonDriversDir(),"inspect-compute-results").toString(),
```

Comment:
Does this work on Windows? If not, it might work if we just remove the python drivers path and assume that "inspect-compute-results" is on the PATH, which is not ideal, but making sure the python code always runs as expected from the command line was non-trivial, so I think it is OK to make use of the PATH.

Reply:
I think it should work as there is a .bat file in the drivers directory, but I will check.

Reply:
I got around to checking this and you are correct: it gives an error.  If I add .bat on to the end of the command then things work.

We could add .bat if the server is running under Windows.  That would avoid the need to assume that the script is on the path.  Are there any downsides to detecting the OS and adding .bat if it is Windows?

Reply:
If that works, then that sounds good enough. Maybe we should have a "runDriver" method for that.

Reply:
Done.

Reply:
...and tested it on Linux and Windows.